stream/src/main/scala/org/apache/pekko/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -640,7 +640,12 @@ import pekko.util.ccompat.JavaConverters._
             else substreamSource.push(elem)
           }
         } catch {
-          case NonFatal(ex) => onUpstreamFailure(ex)
+          case NonFatal(ex) =>
+            decider(ex) match {
+              case Supervision.Resume  => pull(in)
+              case Supervision.Stop    => onUpstreamFailure(ex)
+              case Supervision.Restart => onUpstreamFailure(ex) // TODO implement restart?
```

Comment:
This will be implemented as a result of https://github.com/apache/incubator-pekko/pull/981

Reply:
After I took a look at the code, I think we should handle `Supervision.Restart` as the `Supervision.Resume`,be cause it happens in `SubstreamHandler`

Reply:
> After I took a look at the code, I think we should handle `Supervision.Restart` as the `Supervision.Resume`,be cause it happens in `SubstreamHandler`

Does it matter? The code will be updated in https://github.com/apache/incubator-pekko/pull/981? Currently it's a place holder.

Or do you mean  that in https://github.com/apache/incubator-pekko/pull/981 it should behave as you say?

Reply:
> We should add test for  at least ` Supervision.Resume ` and ` Supervision.Stop` strategy.

These already exist, only the case of throwing exception with `Supervision.restart` being missed